src/components/QuantFile.vue

```diff
@@ -52,8 +52,14 @@ export default {
     },
     id() {
       if (this.quantType.hasMetadata) {
-        // handles barcodes of type ABC-QC and ABC_QC
-        return this.metadata[this.quantType.metadata.idColumn].split(/[-,_]/)[0]
+        // handles barcodes of type ABC-1234-1-QC, ABC-1234-1_QC and ABC-1234-1
+        let barcodeLine = this.metadata[this.quantType.metadata.idColumn]
```

Comment:
You can make this a const. This is usually what we do if the variable is immutable.